### PR TITLE
Update S4 environment and module files

### LINKED ISSUE #1297: Update S4 support
**Description**

The S4 module file and environment files need to be updated to bring support for that system up to current development.

**Requirements**

**Acceptance Criteria (Definition of Done)**

Complete two full GFS (and thus 5 full GDAS) cycles without issue and with similar results to Hera or Orion.

**Dependencies**

None

### PR DESCRIPTION
**Pull request by DavidHuber-NOAA**
**Description**
Updates the S4 module file and environment file to keep up to date with the current develop branch.  Fixes #1297.  Addresses one bug in #1195.
This also increases the memory request for the ediag job.  During testing, that job initially failed.  For the 5 cycles run between 2022051500 - 2022051600, the largest memory footprint was ~26.5GB.

**Type of change**

Please delete options that are not relevant.

- [x] Support update

**How Has This Been Tested?**

  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes

### CODE REVIEW

File: scripts/exgdas_enkf_earc.sh
Comment (WalterKolczynski-NOAA) [inline scripts/exgdas_enkf_earc.sh]: Are you sure this works? My attempts in the past show that once you start defining a base, you have to use a `$` even inside a math expression:
```
Orion-login-4 test/>x='09'
Orion-login-4 test/>a=$(( x + 1 ))
-bash: 09: value too great for base (error token is "09")
Orion-login-4 test/>a=$(( 10#x + 1 ))
-bash: 10#x: value too great for base (error token is "10#x")
Orion-login-4 test/>a=$(( 10#${x} + 1 ))
Orion-login-4 test/>echo $a
10
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: scripts/exgdas_enkf_earc.sh
Comment (DavidHuber-NOAA) [reply scripts/exgdas_enkf_earc.sh]: @WalterKolczynski-NOAA No, it doesn't work.  I tested this with `LOCALARCH=NO`, so I essentially didn't test it.  I'll fix it up and give it a real test.

**Comment by DavidHuber-NOAA** [comment]
My gfsprep job is failing after merging in #1202 at [prep.sh:90](https://github.com/NOAA-EMC/global-workflow/blob/9d11e1e871639523813fcc606aab40f1a6a8103f/jobs/rocoto/prep.sh#L90) as the tcvitals file does not exist.  Should this file always exist, or should it only show up on days with TC activity?  If the former, I will debug; if the latter, I can add a conditional file check before the `mv` command.  I am running on 2022110900, for which there is no reported activity in glopara/com/gfs/prod/syndat/syndat_tcvitals.2022.

File: scripts/exgdas_enkf_earc.sh
Comment (DavidHuber-NOAA) [reply scripts/exgdas_enkf_earc.sh]: Fixed (and tested) by adding `$`s.

**Comment by KateFriedman-NOAA** [comment]
> Should this file always exist, or should it only show up on days with TC activity?

Yes, the `$CDUMP.t$CCz.syndata.tcvitals.tm00` always exists even when there aren't storms (it's size 0).

> I am running on 2022110900, for which there is no reported activity in glopara/com/gfs/prod/syndat/syndat_tcvitals.2022.

Is your copy of `syndat_tcvitals.2022` up-to-date? The tcvitals dump file shows Nicole for that cycle:

```
-bash-4.2$ cat /scratch1/NCEPDEV/global/glopara/dump/gfs.20221109/00/atmos/gfs.t00z.syndata.tcvitals.tm00
NHC  17L NICOLE    20221109 0000 273N 0743W 250 046 0984 1010 0519 28 111 0445 0445 0056 0612 M
```

...and the 2022 syndat copy on Hera also shows Nicole and an INVEST:

```
-bash-4.2$ grep 20221109 /scratch1/NCEPDEV/global/glopara/com/gfs/prod/syndat/syndat_tcvitals.2022
NHC  17L NICOLE    20221109 0000 273N 0743W 250 046 0984 1010 0519 28 111 0445 0445 0056 0612 M -999 -999 -999 0111 72 347N  819W -999 -999 -999 -999 
NHC  17L NICOLE    20221109 0000 273N 0743W 250 046 0984 1010 0519 28 111 0445 0445 0056 0612 M -999 -999 -999 0111 72 347N  819W -999 -999 -999 -999 
NHC  17L NICOLE    20221109 0600 267N 0754W 245 057 0985 1010 0519 31 111 0667 0222 0111 0741 M -999 -999 -999 0111 72 373N  800W -999 -999 -999 -999 
NHC  97L INVEST    20221109 0600 354N 0444W 075 103 1013 1014 0185 18 093 -999 0148 0148 -999 S -999 -999 -999 -999 -9 -99N -999W -999 -999 -999 -999 
NHC  17L NICOLE    20221109 0600 267N 0754W 245 057 0985 1010 0519 31 111 0667 0222 0111 0741 M -999 -999 -999 0111 72 373N  800W -999 -999 -999 -999 
NHC  97L INVEST    20221109 0600 354N 0444W 075 103 1013 1014 0185 18 093 -999 0148 0148 -999 S -999 -999 -999 -999 -9 -99N -999W -999 -999 -999 -999 
NHC  97L INVEST    20221109 1200 364N 0407W 070 144 1006 1008 0148 18 093 -999 0148 0148 -999 S -999 -999 -999 -999 -9 -99N -999W -999 -999 -999 -999 
NHC  17L NICOLE    20221109 1200 266N 0761W 255 046 0985 1010 0519 31 037 0667 0222 0111 0741 M 0037 -999 0037 0093 72 387N  782W -999 -999 -999 -999 
NHC  97L INVEST    20221109 1200 364N 0407W 070 144 1006 1008 0148 18 093 -999 0148 0148 -999 S -999 -999 -999 -999 -9 -99N -999W -999 -999 -999 -999 
NHC  17L NICOLE    20221109 1200 266N 0761W 255 046 0985 1010 0519 31 037 0667 0222 0111 0741 M 0037 -999 0037 0093 72 387N  782W -999 -999 -999 -999 
NHC  17L NICOLE    20221109 1800 265N 0773W 270 051 0985 1010 0519 31 037 0723 0148 0111 0890 M 0037 -999 0037 0093 48 324N  833W -999 -999 -999 -999 
```

Grab the updated syndat files from Hera if needed. I did another update of it earlier this week to get the 2023 file updated.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA I see now that I haven't transferred those syndata files to S4.  I will do that and give this another go.  And thanks for the heads up RE Nicole -- I'll double check the rest of the annual files and make sure they are up to date with Hera as well.

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA The syndat files have been downloaded to the GDA on S4 and the prep steps are running correctly now based on two new GFS cycles.  I will start a cycle test on Hera today.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Approved pending testing

**Pull request merged**
